samples: bluetooth: Increase sync timeout to allow packet loss

rugeGerritsen opened the pull request:
The periodic advertiser samples use a periodic advertising interval in the range of 1 to 1.2 seconds.

Previously the sync timeout of 1.7 seconds would cause the sync to be lost after failing to receive a single periodic advertising packet. This change therefore aims to improve the user experience.

rugeGerritsen committed:
samples: bluetooth: Increase sync timeout to allow packet loss

The periodic advertiser samples use a periodic advertising interval
in the range of 1 to 1.2 seconds.

Previously the sync timeout of 1.7 seconds would cause the sync to be
lost after failing to receive a single periodic advertising packet.
This change therefore aims to improve the user experience.

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

Thalley commented on samples/bluetooth/central_past/src/main.c:
```diff
@@ -293,7 +293,7 @@ int main(void)
 		sync_create_param.sid = per_sid;
 		sync_create_param.skip = 0;
-		sync_create_param.timeout = 0xaa;
+		sync_create_param.timeout = 1000; /* 10 seconds */
```
What we have done in the LE audio samples/tests is that we use the periodic advertising interval to calculate a reasonable timeout. 

That way it will always work with any advertiser regardless of their setup; this change here will still fail for advertisers using an interval of e.g. 12 seconds. 

We've seen several platforms at UPF using a hardcoded timeout like here, and not only is it hard to debug, but it also doesn't really make sense to hardcode it (for example if the interval is 20ms, there's no point in waiting 10 seconds before a timeout)

rugeGerritsen replied:
I agree. That sounds like a better approach. I will update this PR accordingly

rugeGerritsen committed:
bluetooth: Document periodic advertising interval unit

It was not possible to see what unit was used for periodic
advertising interval in the scan recv callback without digging
deep into the code.

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

Thalley commented on samples/bluetooth/direction_finding_connectionless_rx/src/main.c:
```diff
@@ -263,7 +263,7 @@ static void create_sync(void)
 	sync_create_param.sid = per_sid;
 	sync_create_param.skip = 0;
-	sync_create_param.timeout = 0xaa;
+	sync_create_param.timeout = sync_create_timeout_ms / 10;
```
Is this missing a `* 10`?

kartben commented on samples/bluetooth/central_past/src/main.c:
```diff
@@ -293,7 +295,7 @@ int main(void)
 		sync_create_param.sid = per_sid;
 		sync_create_param.skip = 0;
-		sync_create_param.timeout = 0xaa;
+		sync_create_param.timeout = per_adv_interval_ms * 10 / 10; /* 10 attempts */
```
typo?

```suggestion
		sync_create_param.timeout = per_adv_interval_ms * 10; /* 10 attempts */
```

Thalley replied:
Rather it's a missing comment. 

`timeout` is in 10s of milliseconds, so a timeout value of `10` means `100` milliseconds, thus the division of 10 from ms to 10s of ms.

kartben replied:
ah, i see it now. Yeah, comment should be better worded maybe, as this is confusing IMO. Maybe just drop the "10 / 10" altogether and make the comment say "10 attempts (timeout is expressed in 10s of ms)" or something like that?
Anyway, I'm glad I made this a non-blocking comment :)

rugeGerritsen replied:
Maybe we should add a macro BT_GAP_MS_TO_PER_ADV_TIMEOUT to make the code self-documenting?

Thalley replied:
In the LE audio samples and test we something like this
```c
static uint16_t interval_to_sync_timeout(uint16_t interval)
{
	uint32_t interval_ms;
	uint32_t timeout;

	/* Add retries and convert to unit in 10's of ms */
	interval_ms = BT_GAP_PER_ADV_INTERVAL_TO_MS(interval);
	timeout = (interval_ms * PA_SYNC_INTERVAL_TO_TIMEOUT_RATIO) / 10;

	/* Enforce restraints */
	timeout = CLAMP(timeout, BT_GAP_PER_ADV_MIN_TIMEOUT, BT_GAP_PER_ADV_MAX_TIMEOUT);

	return (uint16_t)timeout;
}
```

kartben commented on samples/bluetooth/direction_finding_connectionless_rx/src/main.c:
```diff
@@ -263,7 +263,7 @@ static void create_sync(void)
 	sync_create_param.sid = per_sid;
 	sync_create_param.skip = 0;
-	sync_create_param.timeout = 0xaa;
+	sync_create_param.timeout = sync_create_timeout_ms * 10 / 10; /* 10 attempts */
```
```suggestion
	sync_create_param.timeout = sync_create_timeout_ms * 10; /* 10 attempts */
```

kartben commented on samples/bluetooth/periodic_sync/src/main.c:
```diff
@@ -233,7 +235,7 @@ int main(void)
 		sync_create_param.sid = per_sid;
 		sync_create_param.skip = 0;
-		sync_create_param.timeout = 0xaa;
+		sync_create_param.timeout = per_adv_interval_ms * 10 / 10; /* 10 attempts */
```
```suggestion
		sync_create_param.timeout = per_adv_interval_ms * 10; /* 10 attempts */
```

kartben reviewed (commented):
won't make this a -1 as I feel like I might have missed something obvious and don't want to unnecessarily block, but it looks like this has a typo in timeout calculation?
